### PR TITLE
* added thumb active and inactive icons

### PR DESCRIPTION
**Pull request by jamiecash**
* added thumb active and inactive colors
* added height
* added font size
* added tooltip
* width calculated from size of child widgets
* removed most of the required values and set sensible defaults
* added padding around the widget
* General code changes to comply with latest version of flutter guidelines

### CODE REVIEW

**Comment by jamiecash** [comment]
image
image

**Comment by jamiecash** [comment]
The following additional options along with their default values have been added:

- height = 35,
- fontSize = 16,
- activeTextColor = Colors.white,
- inactiveTextColor = Colors.white,
- activeThumbColor = Colors.white,
- inactiveThumbColor = Colors.white,
- activeTooltip = "",
- inactiveTooltip = "",
- activeThumbIcon,
- inactiveThumbIcon,

File: lib/custom_switch.dart [hunk 1]
```diff
@@ -51,67 +110,67 @@ class _CustomSwitchState extends State<CustomSwitch>
...
+            onTap: () {
+              if (_animationController.isCompleted) {
+                _animationController.reverse();
+              } else {
+                _animationController.forward();
+              }
+              widget.onChanged(!widget.value);
+            },
+            child: Tooltip(
+              message: widget.value ? widget.activeTooltip : widget.inactiveTooltip,
```
Comment (mohak1283) [inline lib/custom_switch.dart:122]: @jamiecash can you share a video/ss with tooltip? how's the switch looking with tooltip? All other changes looks fine to me.

**Review by mohak1283 — CHANGES_REQUESTED** [review]
Minor comment

**Comment by jamiecash** [comment]
image
image

**Comment by jamiecash** [comment]
Added.

**Review by mohak1283 — APPROVED** [review]
LGTM, thanks for your contribution!

**Comment by jamiecash** [comment]
Great, thanks. Could you let me know when this will next be published so
that can update my app and remove my local version.

**Pull request merged**